addons/account/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -1126,7 +1126,26 @@ var StatementModel = BasicModel.extend({
         var formatOptions = {
             currency_id: line.st_line.currency_id,
         };
-        var amount = values.amount !== undefined ? values.amount : line.balance.amount;
+        var amount;
+        switch(values.amount_type) {
+            case 'percentage':
+                amount = line.balance.amount * values.amount / 100;
+                break;
+            case 'regex':
+                var matching = line.st_line.name.match(new RegExp(values.amount_from_label_regex))
+                amount = parseFloat(!matching || matching.length != 2 ? null : matching[1].replace(',', '.').replace(/\.(?=.*\.)/, ""));
+                break;
+            case 'fixed':
+                amount = values.amount;
+                break;
+            case 'remaining':
+                amount = line.balance.amount - line.reconciliation_proposition.slice(-1)[0].amount;
```

Comment:
In quickCreateProposition: function (handle, reconcileModelId)
Change the way it works to match the way it works in python with auto reconcile -> Create first line, then call this._computeLine(line) then create the second line if there is any and finally call _computeLine(line) again.

And you can remove the remaining case